services/dask-sidecar/tests/unit/test__requirements.py

```diff
@@ -0,0 +1,39 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+
+import re
+from pathlib import Path
+from typing import Set, Tuple
+
+import pytest
+
+
+@pytest.fixture(scope="session")
+def requirements_folder(project_slug_dir: Path) -> Path:
+    reqs_dir = project_slug_dir / "requirements"
+    assert reqs_dir.exists()
+    assert any(reqs_dir.glob("_*.txt"))
+    return reqs_dir
+
+
+def test_dask_requirements_in_sync(requirements_folder: Path):
+    """ If this test fails, do update requirements to re-sync all listings """
+
+    REQS_ENTRY_REGEX = re.compile(r"(\w+)==([\.\w]+)")
+    NameVersionTuple = Tuple[str, str]
+
+    def get_reqs(fname: str) -> Set[NameVersionTuple]:
+        return set(REQS_ENTRY_REGEX.findall((requirements_folder / fname).read_text()))
```

Comment:
Could you move these outside of this function?

Reply:
Do you have a special reason? I find it more clear inside of the **only** function it uses it. Short local functions are very common and useful:  I can take advantage of e.g. the scope or  shorter names . Generally, if I move these functions outside, i loose both: I would need to add more parameters and have longer names to avoid collisions and make it more clear in a wider context.